### PR TITLE
KeyValueServiceActorSpec closes #1031

### LINKED ISSUE #1031: Assess/Fix/Implement Ignored Tests in KeyValueServiceActorSpec
The goal here is to look at the tests in the spec, and determine
1. does the test assess something valuable
2. if no, delete it (with an explanation in the PR)
3. if yes, either fix it up so it passes, or reimplement... whatever is more expedient

### PR DESCRIPTION
**Pull request by ruchim**
Three tests are still ignored-- not sure if they are worth keeping.

Small changes to the structure of the case classes of KeyValueService. Hoping to reduce having to pass  around scopes just to gather basic call information like fqn, index and attempt.

### CODE REVIEW

**Comment by Horneth** [comment]
👍

File: services/src/test/scala/cromwell/services/keyvalue/impl/KeyValueServiceActorSpec.scala [hunk 1]
```diff
@@ -18,143 +20,69 @@ class KeyValueServiceActorSpec extends ServicesSpec("KeyValue") {
...
-      //    } yield ()).futureValue
+    //currently the BackendKVStore doesn't accept null values
+    "not be able to store NULL values" ignore {
+//      val kvPut2 = KvPut(KvPair(ScopedKey(wfID, jobKey1, "k2"), None))
+//      val error = new RuntimeException("Failed to find the value associated to key: k2. This key cannot be added to the BackendKVStore.")
+//
+//      (for {
+//        putResult <- (sqlKvServiceActor ? kvPut2).mapTo[KvFailure]
+//        _ = putResult shouldEqual KvFailure(kvPut2, error)
+//      } yield ()).futureValue
```
Comment (scottfrazer) [inline services/src/test/scala/cromwell/services/keyvalue/impl/KeyValueServiceActorSpec.scala]: Is this commented code worth keeping?

File: services/src/test/scala/cromwell/services/keyvalue/impl/KeyValueServiceActorSpec.scala [hunk 2]
```diff
@@ -18,143 +20,69 @@ class KeyValueServiceActorSpec extends ServicesSpec("KeyValue") {
...
-      //      put1 <- ask(kvActor, putB).mapTo[KvPutSuccess]
-      //      _ = put1 shouldEqual KvPutSuccess(putB)
-      //      get1 <- ask(kvActor, getA).mapTo[KvPair]
-      //      _ = get1 shouldEqual KvPair(ScopedKey(callAKey, "k"), Option("value_a"))
-      //      get2 <- ask(kvActor, getB).mapTo[KvPair]
-      //      _ = get2 shouldEqual KvPair(ScopedKey(callBKey, "k"), Option("value_b"))
-      //      _ <- dataAccess.updateWorkflowState(descriptor.id, WorkflowSucceeded)
-      //    } yield ()).futureValue
+
     }
```
Comment (scottfrazer) [inline services/src/test/scala/cromwell/services/keyvalue/impl/KeyValueServiceActorSpec.scala]: Why keep around these empty tests?

**Comment by scottfrazer** [comment]
👍

**Pull request merged**